kvdb-rocksdb/src/iter.rs

```diff
@@ -92,18 +98,25 @@ where
 	/// Creates a new `ReadGuardedIterator` that maps `RwLock<RocksDB>` to `RwLock<DBIterator>`,
 	/// where `DBIterator` iterates over all keys.
 	pub fn new(read_lock: RwLockReadGuard<'a, Option<T>>, col: u32, read_opts: &ReadOptions) -> Self {
-		Self { inner: Self::new_inner(read_lock, |db| db.iter(col, read_opts)) }
+		Self {
+			inner: Self::new_inner(read_lock, |db| db.iter(col, read_opts)),
+			upper_bound_prefix: None,
+		}
 	}
 
 	/// Creates a new `ReadGuardedIterator` that maps `RwLock<RocksDB>` to `RwLock<DBIterator>`,
-	/// where `DBIterator` iterates over keys >= prefix.
-	pub fn new_from_prefix(
+	/// where `DBIterator` iterates over keys which start with the given prefix.
+	pub fn new_with_prefix(
 		read_lock: RwLockReadGuard<'a, Option<T>>,
 		col: u32,
 		prefix: &[u8],
+		upper_bound: Box<[u8]>,
```

Comment:
Small nit, I would have put the calculation to end_prefix (or the call to a function doing the calculation) inside this function to avoid this additional parameter in the api.

Reply:
I agree the current internal API is cumbersome, but we also pass here `&ReadOptions`, so in order to move the `end_prefix` calculation there, a bigger refactoring is needed. I'll leave it as is for now.